onnxscript/rewriter/onnxruntime/fused_matmul_rule_sets.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+from typing import ClassVar
+
+import onnxscript.rewriter.pattern as orp
+
+op = orp.onnxop
+
+
+class FusedMatMulDiv1(orp.RewriteRuleAsClass):
+    """Replaces ``MatMul + Div`` by FusedMatMul."""
+
+    @classmethod
+    def pattern(cls, op, x, y, cst):
+        return op.Div(op.MatMul(x, y), cst)
+
+    @classmethod
+    def check(cls, context, x, y, cst) -> bool:
+        if cst.const_value is None:
+            return False
+        value = cst.const_value.numpy()
+        if value.size > 1:
+            return False
+        return True
+
+    @classmethod
+    def rewrite(cls, op, x, y, cst):
+        value = cst.const_value.numpy()
+        c = float(value[0] if value.shape == (1,) else value)
+        return op.FusedMatMul(x, y, alpha=1 / c, domain="com.microsoft")
+
+
+class FusedMatMulDiv2(orp.RewriteRuleAsClass):
+    """Replaces ``FusedMatMul + Div`` by FusedMatMul."""
+
+    @classmethod
+    def pattern(cls, op, x, y, cst):
+        return op.Div(op.FusedMatMul(x, y, domain="com.microsoft"), cst)
+
+    @classmethod
+    def check(cls, context, x, y, cst) -> bool:
+        if cst.const_value is None:
+            return False
+        if cst.const_value.numpy().size > 1:
+            return False
+        return True
+
+    @classmethod
+    def rewrite(cls, op, x, y, cst):
+        value = cst.const_value.numpy()
+        c = float(value[0] if value.shape == (1,) else value)
+        node = list(x.uses())[0][0]  # noqa: RUF015
+
+        kwargs = {}
+        alpha = node.attributes.get("alpha", None)
+        kwargs["alpha"] = alpha.value / c if alpha else 1.0 / c
+        for name in ["transA", "transB", "transBatchA", "transBatchB"]:
+            att = node.attributes.get(name)
+            if att:
+                kwargs[name] = att.value
+        return op.FusedMatMul(x, y, **kwargs, domain="com.microsoft")
+
+
+class _TransposeMatMulBase(orp.RewriteRuleAsClass):
+    _pos: ClassVar = 1
```

Comment:
Maybe add a comment documenting this? Something like "Indicates the input of MatMul that is transposed."

Reply:
So you have patterns where only one of the two inputs is transposed? Wondering if eventually you would need to support both inputs being transposed.

Reply:
I have but I'll add them in another PR.